### PR TITLE
Include secrets in /config_dump

### PR DESCRIPTION
**Pull request by qfel**
This addresses #166 by adding a "certificates" field to /config_dump output. It generally matches Envoy where a corresponding functionality exists.

Changes:
 - New ConfigDump field
 - Corresponding unit test (mostly to have an example output handy)
 - Support deterministic test cert generation (in particular, use RSA instead of elliptic curves for checked-in test certs to make signatures deterministic)

### CODE REVIEW

File: src/admin.rs [hunk 1]
```diff
@@ -52,6 +57,14 @@ pub struct ConfigDump {
...
     version: BuildInfo,
     config: Config,
+    certificates: Vec<CertsDump>,
+}
+
+#[derive(serde::Serialize, Debug, Clone)]
+pub struct CertsDump {
+    identity: String,
+    leaf: String,
+    chain: Vec<String>,
```
Comment (GregHanson) [inline src/admin.rs:67]: would be good to include [not before/after timestamps](https://github.com/istio/ztunnel/blob/master/src/tls/boring.rs#L115-L116).  I have seen these go expired before while debugging customer env's - it was nice to have that information front and center

**Comment by GregHanson** [comment]
Is there any benefit to including[CertState](https://github.com/istio/ztunnel/blob/master/src/identity/manager.rs#L115)?  `Available` could be inferred by the existence of certs, but do we need to extinguish between `Initializing` and `Unavailable`?

**Comment by GregHanson** [comment]
sample dump output:
```
...
...
  "certificates": [
    {
      "identity": "spiffe://cluster.local/ns/istio-system/sa/ztunnel",
      "leaf": "-----BEGIN CERTIFICATE-----\nMIICezCCAWOgAwIBAgIRAPJhbQo9uXG1NJL8nnA3uyQwDQYJKoZIhvcNAQELBQAw\nGDEWMBQGA1UEChMNY2x1c3Rlci5sb2NhbDAeFw0yMzAzMDYyMDM0MjVaFw0yMzAz\nMDcyMDM2MjVaMAAwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAQXA0bvoSPseHGa\nIfNVIwaSqPdjhklbj2UmFh6dAlFBEnvswtzuCfyyRVXXjq+O7/Ndxp3RljxCUDIr\ncePG53lQo4GiMIGfMA4GA1UdDwEB/wQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcD\nAQYIKwYBBQUHAwIwDAYDVR0TAQH/BAIwADAfBgNVHSMEGDAWgBS7rbt5kI1V5OQK\nsbNHdbzvlQ7TozA/BgNVHREBAf8ENTAzhjFzcGlmZmU6Ly9jbHVzdGVyLmxvY2Fs\nL25zL2lzdGlvLXN5c3RlbS9zYS96dHVubmVsMA0GCSqGSIb3DQEBCwUAA4IBAQDm\nzLhL7xnd9B0K0/P+zi2kHass4TOmHZ2Komv+JVdpYmRciYLtLaDZ3qNG+Hy/nDvZ\nDWwZwn/6UNj2N8Rrp2Ss4JW7sDw/BfhOdsrzeanw2r6V95MazMuk+nSLwC6W/Mjt\ni1j15qMHois8D14SfcmAirU6Pcp8rU/wjMoNuza/F6oZUC/MwhwD/b7o8/hjAHIB\nYYqFgCHy1NOWQfLcZAPruR3KJU7d9iULGw8N6A3S3j+NCqE/rVMp/HA7mxAHLC/B\nQEKN8KkHbdgh5MXyb92scjx5MjKflcq8Yzw+GUXakx2D2Qn4HL+MRnxXoc8OvBYv\nenRmwLJv8B2pUll4Lrdb\n-----END CERTIFICATE-----\n",
      "chain": [
        "-----BEGIN CERTIFICATE-----\nMIIC/DCCAeSgAwIBAgIQQmf2QVj182stXhmALZONXzANBgkqhkiG9w0BAQsFADAY\nMRYwFAYDVQQKEw1jbHVzdGVyLmxvY2FsMB4XDTIzMDMwNjIwMzYxNloXDTMzMDMw\nMzIwMzYxNlowGDEWMBQGA1UEChMNY2x1c3Rlci5sb2NhbDCCASIwDQYJKoZIhvcN\nAQEBBQADggEPADCCAQoCggEBAOiDLcc7wfyqS1mhvAwkMpHAZ9AXCIA/NERBwGlO\nS59i0shbPv49zNl8bMtMbPQi3eXZrhHNRD0Vmtf2kfXuKg9sHhZ1DdIq6hw8nehA\nK6t7HsCEGxWKOFEkRgd1oSk73ZX55v9cGKSbo+/CCq/pDM/aU7r/gRgZqkC5Hj/2\nh55acx8ZA2HmzP29z+Q7HldDGdxtvTfSGgQ4MbutiWSC0T5bx/Q70jwrkv2g9nYW\nM6FMe3dCfZUHXPSGQhl6geFrFsCMDVbmSjXc3qfcW0ccKZbwZHeClp/02KZEFdHO\nBY7dj+w+1sYVi70yiI2+dlWQNI1ACpxe+wmc+hSqrs+TZkcCAwEAAaNCMEAwDgYD\nVR0PAQH/BAQDAgIEMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFLutu3mQjVXk\n5Aqxs0d1vO+VDtOjMA0GCSqGSIb3DQEBCwUAA4IBAQAgILkeH8Bem4sSC3161yKU\nu7kK8jFTQOMWtXHvnxsHXT/rlw+LRYEyg3ej9x3JbbrchnAOW/A+0HK8KW+Rxo+n\n6nQ3KKWyahhC4DJ5RhcoyX+TXCTFdSUoWlvV8DjpMPNr3viMpxt9lw5IKNRUYUIX\nIa8Rjg9L3//eTrkK09GlS80fbKt2AX9+Hg3Dt47ydmTPR0LvAhsu/+LArH0xIGvt\nt7AzYULj1aOznQCuW7WDvj217hOYgPXVifvXrJ0vKqBLouAjRcTjc3RNCwKKD0JG\nIl1pfeLYJABPRl0qg25bO2VQx35h38TAPga02LqdN/fdTQtSvDGEh8M6jpiS+8ho\n-----END CERTIFICATE-----\n"
      ]
    },
    {
      "identity": "spiffe://cluster.local/ns/istio-system/sa/istiod",
      "leaf": "-----BEGIN CERTIFICATE-----\nMIICeTCCAWGgAwIBAgIQT8vm7GJJW6mHVJ33ePmaWDANBgkqhkiG9w0BAQsFADAY\nMRYwFAYDVQQKEw1jbHVzdGVyLmxvY2FsMB4XDTIzMDMwNjIwMzQyNFoXDTIzMDMw\nNzIwMzYyNFowADBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABHsMHHAC5T34ZhXe\nhe6hAxGnfmRPgtRy6/mbXDk545ywkecgfmcncH9RENeP7LbuDGYUMf39uhgpIxJ1\nYEy3BpWjgaEwgZ4wDgYDVR0PAQH/BAQDAgWgMB0GA1UdJQQWMBQGCCsGAQUFBwMB\nBggrBgEFBQcDAjAMBgNVHRMBAf8EAjAAMB8GA1UdIwQYMBaAFLutu3mQjVXk5Aqx\ns0d1vO+VDtOjMD4GA1UdEQEB/wQ0MDKGMHNwaWZmZTovL2NsdXN0ZXIubG9jYWwv\nbnMvaXN0aW8tc3lzdGVtL3NhL2lzdGlvZDANBgkqhkiG9w0BAQsFAAOCAQEACYfB\nQylTyObVvSQwDKq3S/lt+twNTlKuHI8pHXWw4zm0c9IXy4CU6rIEWwJy28esCfNt\n/Fc0iYy05xXMOC/DndXe1fU/32M9vWg9cVICOrS+ffPabIcE/A+aeoS80AzMOntr\nvKX5u0avdEGKg+Mxl4pKlrz32aQqY/0WxlztggTYvTZiem+yFR1F6KQfjOvbF3Ua\nkR8yX3iFWkS43Y+0AHh3DI3A5zIXdW7xjT3xJZ61OjDHn4qTkdJxGLGE9pNWmZ+K\nefaDmaB6auyuR6W4f+hinM8mH6e+qGy+3/HWxARc/3AruSD98LAQWkDI4S63Cw+A\ntQTcYOp6t8NYaFVhjg==\n-----END CERTIFICATE-----\n",
      "chain": [
        "-----BEGIN CERTIFICATE-----\nMIIC/DCCAeSgAwIBAgIQQmf2QVj182stXhmALZONXzANBgkqhkiG9w0BAQsFADAY\nMRYwFAYDVQQKEw1jbHVzdGVyLmxvY2FsMB4XDTIzMDMwNjIwMzYxNloXDTMzMDMw\nMzIwMzYxNlowGDEWMBQGA1UEChMNY2x1c3Rlci5sb2NhbDCCASIwDQYJKoZIhvcN\nAQEBBQADggEPADCCAQoCggEBAOiDLcc7wfyqS1mhvAwkMpHAZ9AXCIA/NERBwGlO\nS59i0shbPv49zNl8bMtMbPQi3eXZrhHNRD0Vmtf2kfXuKg9sHhZ1DdIq6hw8nehA\nK6t7HsCEGxWKOFEkRgd1oSk73ZX55v9cGKSbo+/CCq/pDM/aU7r/gRgZqkC5Hj/2\nh55acx8ZA2HmzP29z+Q7HldDGdxtvTfSGgQ4MbutiWSC0T5bx/Q70jwrkv2g9nYW\nM6FMe3dCfZUHXPSGQhl6geFrFsCMDVbmSjXc3qfcW0ccKZbwZHeClp/02KZEFdHO\nBY7dj+w+1sYVi70yiI2+dlWQNI1ACpxe+wmc+hSqrs+TZkcCAwEAAaNCMEAwDgYD\nVR0PAQH/BAQDAgIEMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFLutu3mQjVXk\n5Aqxs0d1vO+VDtOjMA0GCSqGSIb3DQEBCwUAA4IBAQAgILkeH8Bem4sSC3161yKU\nu7kK8jFTQOMWtXHvnxsHXT/rlw+LRYEyg3ej9x3JbbrchnAOW/A+0HK8KW+Rxo+n\n6nQ3KKWyahhC4DJ5RhcoyX+TXCTFdSUoWlvV8DjpMPNr3viMpxt9lw5IKNRUYUIX\nIa8Rjg9L3//eTrkK09GlS80fbKt2AX9+Hg3Dt47ydmTPR0LvAhsu/+LArH0xIGvt\nt7AzYULj1aOznQCuW7WDvj217hOYgPXVifvXrJ0vKqBLouAjRcTjc3RNCwKKD0JG\nIl1pfeLYJABPRl0qg25bO2VQx35h38TAPga02LqdN/fdTQtSvDGEh8M6jpiS+8ho\n-----END CERTIFICATE-----\n"
      ]
    }
  ]
}
```
does pipe nicely into `openssl`
```
$ k exec -it -n istio-system ds/ztunnel -- curl localhost:15000/config_dump | jq -r '.certificates[0].chain[0]' | openssl x509 -noout -text -in /dev/stdin
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            42:67:f6:41:58:f5:f3:6b:2d:5e:19:80:2d:93:8d:5f
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: O = cluster.local
        Validity
            Not Before: Mar  6 20:36:16 2023 GMT
            Not After : Mar  3 20:36:16 2033 GMT
        Subject: O = cluster.local
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:e8:83:2d:c7:3b:c1:fc:aa:4b:59:a1:bc:0c:24:
                    32:91:c0:67:d0:17:08:80:3f:34:44:41:c0:69:4e:
...
```

**Comment by qfel** [comment]
I have very little higher-level context on the intended use. Should I just match Envoy output?

I'm not sure where the Identity would go and whether you still care about the full PEM, but other than that looks like we could just extract it from existing data.

**Comment by GregHanson** [comment]
> Should I just match Envoy output?

I don't think it has to be exact.  for first pass, I think the main take-aways would be:
 - Identity
 - cert values that be can piped into openssl
 - expiration timestamps

**Review by howardjohn — APPROVED** [review]

**Pull request merged**